test/framework/toolchain.py

```diff
@@ -755,7 +755,7 @@ def test_old_new_iccifort(self):
         self.modtool.purge()
 
         tc = self.get_toolchain('goolfc', version='1.3.12')
-        tc.options.update({'openmp': True})
+        tc.set_options({'openmp': True})
```

Comment:
@akesandgren seems like the problem you were hitting basically boils this, i.e. the test case not calling `set_options` as it's supposed to...

So, the changes w.r.t. the flag corresponding to the `openmp` toolchain option are not necessary, but I feel they make sense regardless.

Let me know what you think.